### PR TITLE
dump events in sandbox

### PR DESCRIPTION
**Pull request by freehan**
Dumps events at the end of each test. Make it easier to debug e2e failure.

### CODE REVIEW

File: pkg/e2e/framework.go [hunk 1]
```diff
@@ -207,6 +207,10 @@ func (f *Framework) RunWithSandbox(name string, t *testing.T, testFunc func(*tes
 		}
 
 		testFunc(t, sandbox)
+
+		if t.Failed() {
+			sandbox.DumpSandboxInfo()
```
Comment (bowei) [inline pkg/e2e/framework.go]: why not always dump cluster events?

File: pkg/e2e/sandbox.go [hunk 2]
```diff
@@ -93,3 +94,40 @@ func (s *Sandbox) MasterUpgraded() bool {
 func (s *Sandbox) MasterUpgrading() bool {
 	return s.f.statusManager.masterUpgrading()
 }
+
+// DumpSandboxInfo dumps information about the sandbox into logs
+func (s *Sandbox) DumpSandboxInfo() {
+	s.dumpSandboxEvents()
```
Comment (bowei) [inline pkg/e2e/sandbox.go]: why have this indirection?

Comment (freehan) [reply pkg/e2e/sandbox.go]: The indirection is in case in the future we need to dump more category of info. (e.g. GCE objects )

**Comment by freehan** [comment]
Ready fo another round

**Pull request merged**